numpy/core/src/multiarray/ctors.c

```diff
@@ -2907,7 +2907,7 @@ _arange_safe_ceil_to_intp(double value)
             "arange: cannot compute length");
         return -1;
     }
-    if (!(NPY_MIN_INTP <= ivalue && ivalue <= NPY_MAX_INTP)) {
+    if (!(NPY_MIN_INTP <= ivalue && ivalue <= (double)NPY_MAX_INTP)) {
```

Comment:
This is strange: shouldn't you need to cast both `NPY_MIN_INTP` and `NPY_MAX_INTP`?